src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -84,14 +87,15 @@ internal partial class SolutionState
                 workspaceVersion: 0,
                 solutionServices: new SolutionServices(workspace),
                 solutionAttributes: solutionAttributes,
-                projectIds: null,
+                projectIds: ImmutableArray<ProjectId>.Empty,
                 idToProjectStateMap: ImmutableDictionary<ProjectId, ProjectState>.Empty,
                 projectIdToTrackerMap: ImmutableDictionary<ProjectId, CompilationTracker>.Empty,
                 filePathToDocumentIdsMap: ImmutableDictionary.Create<string, ImmutableArray<DocumentId>>(StringComparer.OrdinalIgnoreCase),
                 dependencyGraph: ProjectDependencyGraph.Empty,
+#nullable disable warnings // we are passing null here but we're immediately overwriting it -- better to keep the parameter non-null
                 lazyLatestProjectVersion: null)
```

Comment:
💡 Can you pass `null!`?

Reply:
I think we should remove this warning and move the null handling to the private constructor with the correct annotation.

Reply:
So the annotation *is* correct -- that thing really shouldn't be null. What this is doing is a pretty evil hack. I didn't want to annotate lazyLatestProjectVersion as nullable in case another call to the constructor was ever done.